hoki/tests/test_spec.py

```diff
@@ -1,12 +1,164 @@
+
+    def test_type(self):
+        assert type(self.bp) == psp.spectrum.ArraySpectralElement
+
+    def test_name(self):
+        assert self.bp.name == "LICK/LICK.U",\
+               "The filter name is incorrectly imported"
+
+    def test_wavelength(self):
+        assert np.isclose(self.bp.wave,wl).all(),\
+               "The wavelength bins are incorrectly imported"
+
+    def test_throughput(self):
+        assert np.isclose(self.bp.throughput,transmission).all(),\
+                "The throughput is incorrectly imported"
+
+
+class TestLoadBandpass():
+
+    # test that environment variable for pysynphot has been set
+    def test_PYSYN_CDBS_set(self):
+        assert os.environ['PYSYN_CDBS'] != "", "'PYSYN_CDBS' not set"
+
+    # Input-output type checks
+    def test_multi_builtin_filter(self):
+        bp = spec.load_bandpass("acs,hrc,f220w")
+        assert type(bp) == psp.obsbandpass.ObsModeBandpass,\
+               "Wrong output type for multi pysynphot filter"
+
+    def test_single_builtin_filter(self):
+        bp = spec.load_bandpass("johnson,v")
+        assert type(bp) == psp.spectrum.TabularSpectralElement,\
+               "Wrong output type for single pysynphot filter"
+
+    def test_custom_filter(self):
+        bp = spec.load_bandpass(f"{data_path}/LICK.LICK.U.xml")
+        assert type(bp) == psp.spectrum.ArraySpectralElement
+
+    def test_file_not_found(self):
+        with pytest.raises(FileNotFoundError):
+            spec.load_bandpass("f")
+
+    # Tests for checking correct output
+    bp1 = spec.load_bandpass("johnson,v")
+    bp2 = psp.ObsBandpass("johnson,v")
+
+    def test_wave_builtin(self):
+        assert np.isclose(self.bp1.wave, self.bp2.wave).all(),\
+               "The bins are incorrectly created using `load_bandpass`"
+
+    def test_througput_builtin(self):
+        assert np.isclose(self.bp1.throughput, self.bp2.throughput).all(),\
+               "The throughput is incorrect using `load_bandpass`"
+
+    # Tests to check for correct custom filter input
+    bp1 = spec.load_bandpass(f"{data_path}/LICK.LICK.U.xml")
+    bp2 = spec.import_custom_filter(f"{data_path}/LICK.LICK.U.xml")
+
+    def test_wave_custom(self):
+        assert np.isclose(self.bp1.wave, self.bp2.wave).all(),\
+               "The wavelength bins are incorrectly imported using `load_bandpass`"
+
+    def test_throughput_custom(self):
+        assert np.isclose(self.bp1.throughput, self.bp2.throughput).all(),\
+               "The throughput bins are incorrectly imported using `load_bandpass`"
+
+
+class TestBPASStoPSPSpectrum():
+
+    def test_input_type(self):
+        with pytest.raises(TypeError):
+            spec.bpass_to_psp_spectrum(1,[10,10])
+
+    def test_input_pandas(self):
+        sp = spec.bpass_to_psp_spectrum(data.WL,data["6.0"])
+        assert type(sp) == psp.spectrum.ArraySourceSpectrum,\
+            "A panda.Series is not taken as an input"
```

Comment:
would be clearer to say "is not a valid input", but I'm just knit-picking at english now lol

Reply:
changed to "is not recognised as valid input" to accurately represent the test.